### PR TITLE
Replace types string with str.

### PR DESCRIPTION
**Pull request by tswast**
Based on this comment: https://github.com/GoogleCloudPlatform/google-cloud-python/pull/2485#discussion_r83267163

For rtypes, used the command:

    ag -l 'rtype: string' | xargs sed -i .bak 's/rtype: string/rtype: str/g'

For types, used the command:

    ag -l 'type ([^:]+): string' | \
        xargs gsed -r -i.bak -e 's/type ([^:]+): string/type \1: str/g'

Note: [-r for gsed (GNU sed) is needed for group matching](http://superuser.com/a/336819/125262).

### CODE REVIEW

**Comment by jonparrott** [comment]
😆

File: bigquery/google/cloud/bigquery/dataset.py [hunk 1]
```diff
@@ -168,7 +168,7 @@ def created(self):
     def dataset_id(self):
         """ID for the dataset resource.
 
-        :rtype: string, or ``NoneType``
+        :rtype: str, or ``NoneType``
```
Comment (dhermes) [inline bigquery/google/cloud/bigquery/dataset.py]: Might as well drop the `or NoneType` while you're here?

Comment (tswast) [inline bigquery/google/cloud/bigquery/dataset.py]: all of them, or just this one?

**Comment by tswast** [comment]
Thanks. For reviewing. I just added some commits for boolean -> bool, too, based on https://github.com/GoogleCloudPlatform/google-cloud-python/pull/2485#discussion_r83269516

**Comment by dhermes** [comment]
> Thanks. For reviewing. I just added some commits for boolean -> bool, too

GAH! Next time, please "heads up new code" first, then push to the PR. (Since review already finished.) Preference: send two PRs.

**Comment by dhermes** [comment]
@tswast bool stuff looks good too. Ping me (or @jonparrott) when Travis is green?

**Comment by tswast** [comment]
Works for me. (Two PRs next time and pinging for Travis)

**Pull request merged**